src/sagemaker/session.py

```diff
@@ -3463,8 +3464,11 @@ def endpoint_from_production_variants(
         Returns:
             str: The name of the created ``Endpoint``.
         """
+        endpoint_config_name = unique_name_from_base(name)
         if not _deployment_entity_exists(
```

Comment:
Do we still need to check here? We are using a unique name.

Reply:
> I am not sure we chose to reuse the endpoint_config. There might be a limit of number of endpoint configs you can have. I am also not sure if we clean up endpoint configs in our tests. Could you please confirm before shipping this?

seems like we don't delete the endpoints in test.

Reply:
deleted it